Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,13 +1244,37 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis tests the mobile app by executing and running the app binary and analysing the workflows's for vulnerabilities. For example vulnerabilities regarding data storage might be sometimes hard to catch during static analysis, but in dynamic analysis you can easily spot what information is stored persistently and if the information is protected properly. Besides this, dynamic analysis allows the tester to properly identify:
+
+- Business logic flaws
+- Vulnerabilities in the tested environments
+- Weak input validation and bad input/output encoding as they are processed through one or multiple services
+
+Analysis can be assisted by automated tools, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), while assessing an application. An application can be assessed by side-loading it, re-packaging it, or by simply attacking the installed version.
 
 ##### Using Non-Rooted Devices
 
-##### Method Tracing
+Non-rooted devices provide the tester with two benefits:
+
+- Replicate an environment that the application is intended to run on
+- As a tester you don't need to bypass root detection and can use objection to package the app with the Frida Gadget.
+
+In order to dynamically analyze the application, you can rely on [Objection](https://github.com/sensepost/objection) which is leveraging Frida. One way to use objection is to [patch the APK](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---patching-an-apk) with the [frida-gadget](https://www.frida.re/docs/gadget/). Objection communicates then using a Python API with the mobile phone through the installed Gadget application.
```

Comment:
Please remember to make links always conform to the style guide: `[TEXT](URL "NAME")`

Objection -> objection: to be consistent

```suggestion
In order to dynamically analyze the application, you can also rely on [objection](https://github.com/sensepost/objection "objection") which is leveraging Frida. However, in order to be able to use objection on non-rooted devices you have to perform one additional step: [patch the APK](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---patching-an-apk "patching - patching an APK") to include the [Frida gadget](https://www.frida.re/docs/gadget/ "Frida Gadget") library. Objection communicates then using a Python API with the mobile phone through the installed Frida gadget.
```